src/string_bytes.h

```diff
@@ -12,6 +14,51 @@ extern int WRITE_UTF8_FLAGS;
 
 class StringBytes {
  public:
+  class InlineDecoder {
+   public:
+    explicit InlineDecoder(Environment* env) : env_(env), out_(nullptr) {
+    }
+
+    ~InlineDecoder() {
+      if (out_ != out_st_)
+        delete[] out_;
+      out_ = nullptr;
+    }
+
+    inline bool Decode(v8::Handle<v8::String> string,
+                       v8::Handle<v8::Value> encoding,
+                       enum encoding _default) {
```

Comment:
Just a tiny nit and I realize you copied the convention from src/node.cc, but `_default` is a horrible name for a parameter.  Something like `default_encoding` would be better.